### DISCUSSION TITLE
Accessing bin script wrappers in Yarn 3 PnP

### DISCUSSION THREAD

**Discussion by wkhughes** [yarnpkg/berry#4442] [Q&A] [answered]
## NPM Setup
When I was using npm directly with package [ts-proto](https://github.com/stephenh/ts-proto) I was able to access bin scripts as they recommend:
```
protoc --plugin=protoc-gen-ts_proto=.\node_modules\.bin\protoc-gen-ts_proto.cmd --ts_proto_out=. ./simple.proto
```

Specifically the `".\node_modules\.bin\protoc-gen-ts_proto.cmd"` part is what I need access to.

## Attempted Yarn Setup
I'm trying to figure out how to replicate this under Yarn 3 PnP. So far I've tried:
1. Unplugging ts-proto so I have access to it outside of a zip file
2. In a script run `yarn bin protoc-gen-ts_proto` to get the bin path in the unplugged folder

However, this only gives me access to the file that begins with `#!/usr/bin/env node` and not the script wrappers that npm provides for every bin script to invoke that file under node (on Windows, these are a .cmd, .ps1 and shell script).

Any ideas on how I could get access to the .cmd script? Thanks!

**Comment by wkhughes** [comment, marked as answer]
I found it's possible to use the `BERRY_BIN_FOLDER` environment variable within a script under `yarn run` to have access to the temp folder with the bin files and wrapper (.cmd) scripts.

Specifically within a powershell script I do something like this:
```
$tsProtoPluginPath = Join-Path $Env:BERRY_BIN_FOLDER "protoc-gen-ts_proto.cmd"
```
Which I then pass to protoc.

**Comment by arcanis** [comment]
Getting the path to the shellscripts isn't a use case we officially support. At the moment those shellscripts are created only for the duration of the `yarn run` commands, and are getting cleaned up afterwards.
